usb/esp_modem_usb_dte/esp_modem_usb_api_target.cpp

```diff
@@ -14,9 +17,20 @@ static const char *TAG = "modem_usb_api_target";
 namespace esp_modem {
 std::shared_ptr<DTE> create_usb_dte(const dte_config *config)
 {
+    if (config == nullptr) {
+        return nullptr;
+    }
+
+    // *INDENT-OFF*
     TRY_CATCH_RET_NULL(
-        auto term = create_usb_terminal(config);
-        return std::make_shared<DTE>(config, std::move(term));
+        const struct esp_modem_usb_term_config *usb_config = (struct esp_modem_usb_term_config *)(config->extension_config);
```

Comment:
This doesn't look very much like `C++` style. Sorry, probably missed in the previous review. Suggest something like:
```suggestion
        auto *usb_config = static_cast<struct esp_modem_usb_term_config *>(config->extension_config);
```

cc @euripedesrocha (if you have a better idea, but the `extension_config` could be also used from plain C interface, so I think the `void*` makes sense here)